cmd/wstail/tail.go

```diff
@@ -151,30 +128,9 @@ type dockerLogMessage struct {
 	Log string `json:"log"`
 }
 
-// newCloseNotifierLoop spawns a goroutine that reads from the client.
-// The loop terminates once a message from the client has been received.
-// The only expected message the close message although the goroutine does not validate this fact.
-// Returns a channel that will be closed if client disconnects.
-func newCloseNotifierLoop(ws *websocket.Conn) chan struct{} {
-	notifierC := make(chan struct{})
-	go func() {
-		var err error
-		for err == nil {
-			var msg []byte
-			_, msg, err = ws.ReadMessage()
-			if err == nil {
-				log.Infof("recv: %s", msg)
-			}
-		}
-		log.Infof("closing connection with %v", err)
-		close(notifierC)
-	}()
-	return notifierC
-}
-
-// newMessagePump spawns a goroutine to handle messages from the tailing process group.
+// makeOutputChannel spawns a goroutine to handle messages from the process group.
 // Returns a channel where the received messages are sent to.
-func newMessagePump(r io.Reader, history io.ReadCloser) chan string {
+func makeOutputChannel(r io.Reader, history io.ReadCloser) chan string {
```

Comment:
Not sure I agree with the name change - it reads as if it only created and returned a channel.